modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jackson/databind/ser/ExtendedEntityPropertyFilter.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.vulcan.internal.jackson.databind.ser;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.PropertyFilter;
+import com.fasterxml.jackson.databind.ser.PropertyWriter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Javier de Arcos
+ */
+public class ExtendedEntityPropertyFilter
+	extends SimpleBeanPropertyFilter implements PropertyFilter {
+
+	public ExtendedEntityPropertyFilter(
+		PropertyFilter baseFilter, Set<String> filteredFields) {
+
+		_baseFilter = baseFilter;
+		_filteredFields = new HashSet<>(filteredFields);
+	}
+
+	@Override
+	public void serializeAsField(
+			Object object, JsonGenerator jsonGenerator,
+			SerializerProvider serializerProvider,
+			PropertyWriter propertyWriter)
+		throws Exception {
+
+		if (!include(propertyWriter)) {
+			return;
+		}
+
+		_baseFilter.serializeAsField(
+			object, jsonGenerator, serializerProvider, propertyWriter);
+	}
+
+	@Override
+	protected boolean include(PropertyWriter writer) {
+		return !_filteredFields.contains(writer.getName());
+	}
+
+	private final PropertyFilter _baseFilter;
```

Comment:
this should be propertyFilter I think

Reply:
I've called it baseFilter trying to explain that this is a decorator and the base entity will be call as part of the logic but it's true that propertyFilter may be preferable...

Reply:
Renamed to _basePropertyFilter